inference/core/workflows/core_steps/models/foundation/yolo_world.py

```diff
@@ -187,25 +197,25 @@ async def run_remotely(
                 confidence=confidence,
             )
             predictions.extend(sub_batch_predictions)
-        return self._post_process_result(image=images, predictions=predictions)
+        results = self._post_process_result(
+            images=non_empty_images, predictions=predictions
+        )
+        return images.align_batch_results(
+            results=results, null_element={"predictions": None}
+        )
 
     def _post_process_result(
         self,
-        image: List[dict],
+        images: List[WorkflowImageData],
         predictions: List[dict],
     ) -> List[Dict[str, Union[sv.Detections, Any]]]:
-        detections = convert_to_sv_detections(predictions)
-        for p, d in zip(predictions, detections):
-            p["predictions"] = d
-        predictions = attach_prediction_type_info(
+        predictions = convert_inference_detections_batch_to_sv_detections(predictions)
```

Comment:
within inference project do we have another structure which should be converted to sv.Detections?

Reply:
not at this moment